couchdb/tests/server/qs.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+#
+import unittest
+
+from functools import partial
+
+from couchdb import json
+from couchdb.server import BaseQueryServer, SimpleQueryServer
+from couchdb.server import exceptions
+from couchdb.util import StringIO
+
+
+class BaseQueryServerTestCase(unittest.TestCase):
+
+    def test_set_config_option(self):
+        server = BaseQueryServer(foo='bar')
+        self.assertTrue('foo' in server.config)
+        self.assertEqual(server.config['foo'], 'bar')
+
+    def test_config_option_handler(self):
+        class CustomServer(BaseQueryServer):
+            def config_foo(self, value):
+                self.config['baz'] = value
+        server = CustomServer(foo='bar')
+        self.assertTrue('foo' not in server.config)
+        self.assertTrue('baz' in server.config)
+        self.assertEqual(server.config['baz'], 'bar')
+
+    def test_process_request(self):
+        server = BaseQueryServer()
+        server.commands['foo'] = lambda s, x: x == 42
+        self.assertTrue(server.process_request(['foo', 42]))
+
+    def test_process_request_ddoc(self):
+        server = BaseQueryServer()
+        server.commands['foo'] = lambda s, x: x == 42
+        self.assertTrue(server.process_request(['foo', 42]))
```

Comment:
The content of the function `test_process_request` and `test_process_request_ddoc` is same!
Ref to original code: https://github.com/kxepal/couchdb-python/blob/91455455a9e172c75b56ae7e1185d6641b005594/couchdb/tests/server/qs.py#L180-L188

So i start to trace the code and found that [`BaseQueryServer._commands_ddoc`](https://github.com/djc/couchdb-python/blob/60b85555968483bfe79f7580f73097d8dfd571bd/couchdb/server/__init__.py#L48) is a poor orphan.
Even `SimpleQueryServer` [do not use this var](https://github.com/djc/couchdb-python/blob/60b85555968483bfe79f7580f73097d8dfd571bd/couchdb/server/__init__.py#L337-L349). :sob: